Implement Cron-Cluster

catrielmuller opened the pull request:
Support for Cluster Mode, for not run the same job in all server of the cluster. 
For example if i have a job for resize a image i dont want a run the same task in all server cos is a waste of resources and in some case its impossible running the same job two times on the same time.
This PR use the configuration of the Trailpack-cache for automatically switch to cron-cluster if you have a redis server configured.
Take a look.
Cheers.

catrielmuller committed:
Implement Cron-Cluster

catrielmuller committed:
Fix EsLint Problems

jaumard commented:
Hi @catrielmuller ! Thanks for this ! Very interesting PR ! 
One things that is not needed in my opinion. Why the `trailpack-cache` dependency ? It's a shame to add this dependency between 2 trailpacks not related to each other. 
We can just add a redis config into `config/cron` like this : 

```
{
  defaultTimeZone: 'Europe/Paris', // Default timezone use for tasks
  cluster: {
    name: 'my-redis-store',
    type: 'redis',
    host: 'localhost',
    auth_pass: ''
    db: 0,
    ttl: 600 // Default TTL
  },
  jobs: {
    myJob: {
      schedule: '* * * * * *',
      onTick: app => {
        app.log.info('I am ticking')
      },
      onComplete: app => {
        app.log.info('I am done')
      },
      start: true, // Start task immediately
      timezone: 'Europe/Paris' // Custom timezone
    }
  }
}
```

Like this users are not force to install trailpack-cache only to have cluster mode. Can you do the modifications and change the README ? 

PS: there a problem with deps on node 4 look at travis build.

catrielmuller committed:
- Remove Dep for trailcache
- Fix cron deps problem in Node 4
- Fix Readme with the configuration of the cluster
- Fix Readme with the basic configuration (the jobs need to include inside of the jobs attr)

catrielmuller commented:
Yeap, you are right. 
Take a look the new changes.

jaumard commented:
Much better :D did you test it ? It's working ?  Don't have time to test to see if it's works ^^ I trust you :)

catrielmuller commented:
@jaumard I recorded two videos for you can see the workarround.

Trailpack-Cron in a cluster without Cluster Mode
[![asciicast](https://asciinema.org/a/85765.png)](https://asciinema.org/a/85765)

Trailpack-Cron in a cluster with Cluster Mode
[![asciicast](https://asciinema.org/a/85766.png)](https://asciinema.org/a/85766)

jaumard commented:
Very nice ! Thanks for the demo @catrielmuller ! And thanks for this :)
